bittensor/utils/balance.py

```diff
@@ -258,9 +258,9 @@ def from_float(amount: float):
         return Balance(rao)
 
     @staticmethod
-    def from_tao(amount: float):
+    def from_tao(amount: Union[int, float]):
         """
-        Given tao (float), return Balance object with rao(int) and tao(float), where rao = int(tao*pow(10,9))
+        Given tao (int | float), return Balance object with rao(int) and tao(float), where rao = int(tao*pow(10,9))
 
         Args:
             amount: The amount in tao.
```

Comment:
While `from_tao` and `from_rao` accept ints or floats, I think the intention of keeping them `tao: float` and `rao: int` is a good idea.

Reply:
In general, I added this because in different parts of the code for both methods we pass both  types (int and float).
Let's return it as it was and more strictly control which types we will pass.